src/cmap/connect.ts

```diff
@@ -182,7 +184,12 @@ function performInitialHandshake(
           );
         }
         provider.auth(authContext, err => {
-          if (err) return callback(err);
+          if (err) {
+            if (err instanceof MongoError) {
```

Comment:
This has been updated to use `needsRetryableWriteLabel` and that code has been refactored to not take the `maxWireVersion` into account anymore, since the configure fail point error does not automatically add the label in 4.4+ servers. So the code basically just looks to see if the label is there or not and ignores the wire version.

Reply:
Sorry I LGTM-ed but this looks outstanding still
@durran What're your thoughts on point 4? Copying the existing tests and changing the error code seems like it could "just work" 🤞  Maybe we merge this and open a follow up PR for the sake of ease of review, thoughts?